src/box/lua/config/instance_config.lua

```diff
@@ -1246,6 +1246,21 @@ return schema.new('instance_config', schema.record({
                                 type = 'string',
                             }),
                         }),
+                        lua_eval = schema.array({
+                            items = schema.scalar({
+                                type = 'string',
+                            }),
+                        }),
+                        lua_call = schema.array({
+                            items = schema.scalar({
+                                type = 'string',
+                            }),
+                        }),
```

Comment:
What if something different from `null` and `[all]` is set here? Is it worth to add a validation callback to get an error earlier if something different is set? We can remove this validation when (if) `lua_call` will gain more possible values.

Reply:
Added verification and according tests to `instance_config.lua`.

Reply:
How about the following solution?

```diff
diff --git a/src/box/lua/config/instance_config.lua b/src/box/lua/config/instance_config.lua
index 387a4d7c37..c8303e4001 100644
--- a/src/box/lua/config/instance_config.lua
+++ b/src/box/lua/config/instance_config.lua
@@ -1252,11 +1252,13 @@ return schema.new('instance_config', schema.record({
                         lua_call = schema.array({
                             items = schema.scalar({
                                 type = 'string',
+                                allowed_values = {'all'},
                             }),
                         }),
                         sql = schema.array({
                             items = schema.scalar({
                                 type = 'string',
+                                allowed_values = {'all'},
                             }),
                         }),
                     }),
@@ -1316,11 +1318,13 @@ return schema.new('instance_config', schema.record({
                         lua_call = schema.array({
                             items = schema.scalar({
                                 type = 'string',
+                                allowed_values = {'all'},
                             }),
                         }),
                         sql = schema.array({
                             items = schema.scalar({
                                 type = 'string',
+                                allowed_values = {'all'},
                             }),
                         }),
                     }),
@@ -1334,35 +1338,6 @@ return schema.new('instance_config', schema.record({
                 }),
             }),
         }),
-    }, {
-        validate = function(credentials, w)
-            -- Check that only 'all' is provided for `lua_call` and `sql`.
-            if credentials == nil then
-                return
-            end
-            local function check(user_or_role, config)
-                for name, def in pairs(config or {}) do
-                    for _, priv in ipairs(def.privileges or {}) do
-                        local n, val = next(priv.lua_call or {})
-                        if n ~= nil and (next(priv.lua_call, n) ~= nil or
-                                         val ~= 'all') then
-                            w.error('Bad `lua_call` configuration for %s ' ..
-                                    '%q, only "all" is allowed.', user_or_role,
-                                    name)
-                        end
-                        n, val = next(priv.sql or {})
-                        if n ~= nil and (next(priv.sql, n) ~= nil or
-                                         val ~= 'all') then
-                            w.error('Bad `sql` configuration for %s ' ..
-                                    '%q, only "all" is allowed.', user_or_role,
-                                    name)
-                        end
-                    end
-                end
-            end
-            check('user', credentials.users)
-            check('role', credentials.roles)
-        end,
     }),
     app = schema.record({
         file = schema.scalar({
diff --git a/test/config-luatest/instance_config_schema_test.lua b/test/config-luatest/instance_config_schema_test.lua
index 3ffcd38290..d94003bfbe 100644
--- a/test/config-luatest/instance_config_schema_test.lua
+++ b/test/config-luatest/instance_config_schema_test.lua
@@ -965,8 +965,9 @@ g.test_credentials = function()
             },
         },
     }
-    local err = '[instance_config] credentials: Bad `lua_call` configuration' ..
-                ' for role "myrole", only "all" is allowed.'
+    local path = 'credentials.roles.myrole.privileges[1].lua_call[1]'
+    local err = ('[instance_config] %s: Got myfunc, but only the following ' ..
+        'values are allowed: all'):format(path)
     t.assert_error_msg_equals(err, function()
         instance_config:validate(iconfig)
     end)
@@ -987,8 +988,9 @@ g.test_credentials = function()
             },
         },
     }
-    err = '[instance_config] credentials: Bad `sql` configuration ' ..
-                'for user "myuser", only "all" is allowed.'
+    local path = 'credentials.users.myuser.privileges[1].sql[1]'
+    local err = ('[instance_config] %s: Got myfunc, but only the following ' ..
+        'values are allowed: all'):format(path)
     t.assert_error_msg_equals(err, function()
         instance_config:validate(iconfig)
     end)
```

It looks simpler and reports all the needed information in the error message.

Reply:
Applied.